packages/tracer/src/Tracer.ts

```diff
@@ -435,9 +444,18 @@ class Tracer extends Utility implements TracerInterface {
           let result;
           try {
             result = await originalMethod.apply(this, [...args]);
-            tracerRef.addResponseAsMetadata(result, originalMethod.name);
+            if (options?.captureResponse ?? true) {
+              tracerRef.addResponseAsMetadata(result, originalMethod.name);
+            }
+
           } catch (error) {
-            tracerRef.addErrorAsMetadata(error as Error);
+            if (options?.captureError ?? true) {
+              tracerRef.addErrorAsMetadata(error as Error);
+            } else {
+              // Though we aren't adding the error as metadata, we should still
+              // mark the segment as having an error.
+              tracerRef.getSegment().addErrorFlag();
```

Comment:
Please, let's keep the PR focused to this feature and let's discuss this change in a separate issue. After discussing this, if we reach an agreement & we decide to make the change, we'll open a PR.

Reply:
Ok. I'll remove the error capture feature to reduce the scope of this PR to just capturing results.